samples/subsys/ipc/openamp/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@ cmake_minimum_required(VERSION 3.20.0)
 # SPDX-License-Identifier: Apache-2.0
 #
 
-set(REMOTE_ZEPHYR_DIR ${CMAKE_CURRENT_BINARY_DIR}/openamp_remote-prefix/src/openamp_remote-build/zephyr)
+set(REMOTE_ZEPHYR_DIR ${CMAKE_CURRENT_BINARY_DIR}/../openamp_remote/zephyr)
```

Comment:
Somewhat unfortunate. It has to tightly coupled expectation on where the include dir of the second remote image is located.

It's knowledge like this we should be able to share in a generic way through sysbuild.
Will do some thinking on designing a better approach.

Reply:
Sounds good. Maybe we could expose this with a variable?

Reply:
I have wanted to fix this for some time: https://github.com/zephyrproject-rtos/zephyr/pull/49286
not solving this exact situation but laying down a foundation so we can easier handle a case like this.

Reply:
`zephyr_get` looks like it would solve this issue more generically. For the time being, will leave the current implementation as is, until that infrastructure is available.

Reply:
`zephyr_get()` PR #49674 has been merged.
Could you try to update this PR, and see if more functionality is needed.

I assume we need to split the `ExternalZephyrProject_Add()` into two stages.
Stage 1 which will handle the request for adding a Zephyr image, and then a later stage 2 that will invoke the CMake configuration of the image list.

This will allow us to know all images before running CMake, and thus allow us to share location of each image for CMake.

I don't believe this PR should be blocked by such improvement, as it is more important to transitionto sysbuild and gain knowledge of improvements to work on.

Reply:
I tested `zephyr_get()` with this PR, and I agree that it seems like we need a split like you described. Currently, the main sysbuild application must be aware of the build location of the remote application, but the primary sysbuild application is configured in the call to `ExternalZephyrProject_Add()`. This call occurs prior to the secondary application(s) being added via `sysbuild.cmake`

Maybe we could record all image names prior to running CMake configuration, and write these image names into a CMake cache file that all images can load variables from. We could then add this cache file into the list of sources used by `zephyr_get()`